Fix buildsystem and tests

wjwwood opened the pull request:
I have several fixes here to both the build system and unit tests.

Also I have fixes for pcl-1.7 related changes and pcl_ros changes.

hershwg commented:
Wow, thanks for all this.

It's a shame that catkin_download_test_data needs to be told the
destination for the file every time. That really bloats up the file with a
lot of repetition.
On Jul 10, 2013 12:11 AM, "William Woodall" notifications@github.com
wrote:

> I have several fixes here to both the build system and unit tests.
> 
> ## Also I have fixes for pcl-1.7 related changes and pcl_ros changes.
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/wjwwood/navigation fix_buildsystem_and_tests
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/ros-planning/navigation/pull/86
> Commit Summary
> - amcl: Fixup CMake and unit tests
> - base_local_planner: CMake clean up
> - More fixes for building, installing and testing
> - Fixes to comply with pcl-1.7 changes
> 
> File Changes
> - _M_ amcl/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-0(60)
> - _M_ amcl/cfg/AMCL.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-1(8)
> - _M_ amcl/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-2(12)
> - _M_ amcl/test/basic_localization.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-3(113)
> - _M_ amcl/test/basic_localization_stage.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-4(4)
> - _M_ amcl/test/global_localization_stage.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-5(4)
> - _M_ amcl/test/rosie_multilaser.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-6(4)
> - _M_ amcl/test/set_initial_pose.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-7(4)
> - _M_ amcl/test/set_pose.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-8(10)
> - _M_ amcl/test/small_loop_crazy_driving_prg.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-9(4)
> - _M_ amcl/test/small_loop_prf.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-10(4)
> - _M_ amcl/test/texas_greenroom_loop.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-11(4)
> - _M_ amcl/test/texas_willow_hallway_loop.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-12(4)
> - _M_ base_local_planner/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-13(10)
> - _M_ base_local_planner/cfg/BaseLocalPlanner.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-14(10)
> - _M_ base_local_planner/cfg/LocalPlannerLimits.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-15(10)
> - _M_
>   base_local_planner/include/base_local_planner/map_grid_cost_point.hhttps://github.com/ros-planning/navigation/pull/86/files#diff-16(2)
> - _M_ base_local_planner/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-17(4)
> - _M_ base_local_planner/src/local_planner_limits/**init**.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-18(55)
> - _M_ base_local_planner/src/map_grid_visualizer.cpphttps://github.com/ros-planning/navigation/pull/86/files#diff-19(6)
> - _M_ carrot_planner/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-20(7)
> - _M_ carrot_planner/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-21(2)
> - _M_ clear_costmap_recovery/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-22(19)
> - _M_ costmap_2d/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-23(70)
> - _M_ costmap_2d/cfg/Costmap2D.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-24(15)
> - _M_ costmap_2d/cfg/GenericPlugin.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-25(8)
> - _M_ costmap_2d/cfg/InflationPlugin.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-26(10)
> - _M_ costmap_2d/cfg/ObstaclePlugin.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-27(10)
> - _M_ costmap_2d/cfg/VoxelPlugin.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-28(10)
> - _M_ costmap_2d/include/costmap_2d/costmap_math.hhttps://github.com/ros-planning/navigation/pull/86/files#diff-29(8)
> - _M_ costmap_2d/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-30(46)
> - _M_ costmap_2d/plugins/voxel_layer.cpphttps://github.com/ros-planning/navigation/pull/86/files#diff-31(6)
> - _M_ costmap_2d/src/observation_buffer.cpphttps://github.com/ros-planning/navigation/pull/86/files#diff-32(17)
> - _M_ dwa_local_planner/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-33(21)
> - _M_ dwa_local_planner/cfg/DWAPlanner.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-34(15)
> - _M_ dwa_local_planner/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-35(24)
> - _M_ dwa_local_planner/src/dwa_planner.cpphttps://github.com/ros-planning/navigation/pull/86/files#diff-36(11)
> - _M_ fake_localization/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-37(2)
> - _M_ fake_localization/static_odom_broadcaster.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-38(18)
> - _M_ global_planner/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-39(14)
> - _M_ global_planner/cfg/GlobalPlanner.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-40(2)
> - _M_ map_server/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-41(30)
> - _M_ map_server/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-42(2)
> - _M_ map_server/test/consumer.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-43(17)
> - _M_ move_base/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-44(7)
> - _M_ move_base/cfg/MoveBase.cfghttps://github.com/ros-planning/navigation/pull/86/files#diff-45(3)
> - _M_ move_base/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-46(45)
> - _M_ move_base/scripts/subtopic_forwarder.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-47(57)
> - _M_ move_base/scripts/subtopic_forwarder_node.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-48(21)
> - _M_ move_base/scripts/warner.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-49(32)
> - _M_ move_slow_and_clear/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-50(6)
> - _M_ nav_core/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-51(5)
> - _M_ navfn/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-52(12)
> - _M_ navfn/include/navfn/potarr_point.hhttps://github.com/ros-planning/navigation/pull/86/files#diff-53(2)
> - _M_ navfn/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-54(22)
> - _M_ navfn/src/navfn_ros.cpphttps://github.com/ros-planning/navigation/pull/86/files#diff-55(7)
> - _M_ robot_pose_ekf/scripts/wtf.pyhttps://github.com/ros-planning/navigation/pull/86/files#diff-56(6)
> - _M_ voxel_grid/CMakeLists.txthttps://github.com/ros-planning/navigation/pull/86/files#diff-57(2)
> - _M_ voxel_grid/package.xmlhttps://github.com/ros-planning/navigation/pull/86/files#diff-58(2)
> 
> Patch Links:
> - https://github.com/ros-planning/navigation/pull/86.patch
> - https://github.com/ros-planning/navigation/pull/86.diff

hershwg commented on amcl/test/basic_localization.py:
```diff
@@ -16,67 +12,68 @@
 
+
 class TestBasicLocalization(unittest.TestCase):
+    def setUp(self):
```
Why did this entire file need to be re-indented like this?

wjwwood replied:
There were different indentions in the file, which is bad for python, so I reformatted it to four spaces (PEP8).

hershwg commented:
In general I prefer if format changes are saved up for separate pull requests (or at least separate commits), so it is easy for me to look at the substantive changes and think about them.  As is, this pull request seems to be about 50% formatting changes and 50% substantive changes.

hershwg commented:
I'm waiting to merge this on the release of debians for pcl_ros 1.1.0 and pcl_conversions.  We are using hydro-devel branch of navigation in our project and don't want to have to compile pcl_ros etc from source.

wjwwood commented:
> It's a shame that catkin_download_test_data needs to be told the
> destination for the file every time. That really bloats up the file with a
> lot of repetition.

That's true, it has a default location, but it the default wouldn't work as well in this case.

> In general I prefer if format changes are saved up for separate pull requests (or at least separate commits), so it is easy for me to look at the substantive changes and think about them. As is, this pull request seems to be about 50% formatting changes and 50% substantive changes.

Sorry I started out like that, but things got mixed up, I'll try to split things up.

> I'm waiting to merge this on the release of debians for pcl_ros 1.1.0 and pcl_conversions. We are using hydro-devel branch of navigation in our project and don't want to have to compile pcl_ros etc from source.

I released them last night, but there appears to be issues on the farm, I'll be looking into that today.

hershwg commented:
The formatting thing is just a preference of mine, it's not something you
need to rework the current pull request for.  I don't want to make more
work for you, I was just whining.

On Wed, Jul 10, 2013 at 10:14 AM, William Woodall
notifications@github.comwrote:

> It's a shame that catkin_download_test_data needs to be told the
> destination for the file every time. That really bloats up the file with a
> lot of repetition.
> 
> That's true, it has a default location, but it the default wouldn't work
> as well in this case.
> 
> In general I prefer if format changes are saved up for separate pull
> requests (or at least separate commits), so it is easy for me to look at
> the substantive changes and think about them. As is, this pull request
> seems to be about 50% formatting changes and 50% substantive changes.
> 
> Sorry I started out like that, but things got mixed up, I'll try to split
> things up.
> 
> I'm waiting to merge this on the release of debians for pcl_ros 1.1.0 and
> pcl_conversions. We are using hydro-devel branch of navigation in our
> project and don't want to have to compile pcl_ros etc from source.
> 
> I released them last night, but there appears to be issues on the farm,
> I'll be looking into that today.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-planning/navigation/pull/86#issuecomment-20757847
> .

wjwwood commented:
@hershwg no I totally agree, I planned to do a layered pull request, but I just wanted to get the changes commit and proposed last night. I really would like this stuff to get reviewed, because I may have made mistakes, and that is all the easier more accurate with more granular changes.

wjwwood committed:
amcl: remove roslib and run_depend on rostest

I also alphabetized some things.

wjwwood committed:
amcl: Move include_directories above executables

wjwwood committed:
amcl: use catkin_download_data and fix tests

wjwwood committed:
amcl: [style] PEP8 python code and cleanup

This includes removing roslib.load_manifest calls

wjwwood committed:
blp: do not use SYSTEM in include_directories

wjwwood committed:
blp: remove redundant call to catkin_python_setup

wjwwood committed:
blp: remove dependency on roslib

As far as I can tell this is no longer necessary.

wjwwood committed:
blp: [style] PEP8 cfg files

I also removed references to roslib, this was not
a required action as dynamic reconfigure brings
roslib into the environment anyways.

wjwwood committed:
blp: fix deprecation warning from pcl-1.7

wjwwood committed:
blp: Fix pcl-1.7 errors by using pcl_conversions

pcl_conversions is intentionally not listed as a
run_depend and not put in the CATKIN_DEPENDS of
catkin_package(..) because it is only used in a
cpp file in base_local_planner.

wjwwood committed:
carrot_planner: remove old dependency on roslib

wjwwood committed:
carrot_planner: install headers

wjwwood committed:
ccr: do not use SYSTEM in include_directories

wjwwood committed:
clear_costmap_recovery: install headers

wjwwood committed:
ccr: [style] alphabetize elements in CMake

wjwwood committed:
costmap_2d: move catkin_package above targets

wjwwood committed:
costmap_2d: do not use SYSTEM in include_dirs

wjwwood committed:
costmap_2d: find_package and use Boost

wjwwood committed:
costmap_2d: [style] alphabetization of elements

wjwwood committed:
costmap_2d: remove dependency on roslib

wjwwood committed:
costmap_2d: update dependencies, add and remove

I did this by going through all of the headers
looking for uses of headers from other packages
and then I updated the package.xml and CMake files
to reflect them.

wjwwood committed:
costmap_2d: [style] PEP8 cfg files

wjwwood committed:
costmap_2d: [style] put space after #include

wjwwood committed:
costmap_2d: fix deprecation warning from pcl-1.7

wjwwood committed:
costmap_2d: update to reflect pcl-1.7 changes

wjwwood committed:
dwalp: use REQUIRED on Eigen and PCL find_package

wjwwood committed:
dwalp: remove references to roslib

wjwwood committed:
dwalp: [style] alphabetize elements

wjwwood committed:
dwalp: install headers

wjwwood committed:
dwalp: [style] PEP8 cfg files

wjwwood committed:
dwalp: update to reflect pcl-1.7 changes

wjwwood committed:
fake_loc: add catkin and boost include directories

wjwwood committed:
fake_loc: remove unused roslib reference

wjwwood committed:
fake_loc: [style] PEP8 python file

wjwwood committed:
global_planner: install headers

wjwwood committed:
global_planner: [style] combine install CMake rule

wjwwood committed:
global_planner: [style] remove import * in cfg

wjwwood committed:
map_server: do not explicitly set CMAKE_BUILD_TYPE

wjwwood committed:
map_server: remove references to roslib

wjwwood committed:
map_server: include catkin_INCLUDE_DIRS

wjwwood committed:
map_server: [style] cleanup CMake install rules

wjwwood committed:
map_server: [style] PEP8 python file

wjwwood committed:
move_base: REQUIRED and use Eigen in CMake

wjwwood committed:
move_base: remove roslib references

wjwwood committed:
move_base: [style] alphabetization of elements

wjwwood committed:
move_base: [style] PEP8 changes

wjwwood committed:
move_slow_and_clear: install headers

wjwwood committed:
nav_core: [style] fixup CMake install rule

wjwwood committed:
navfn: [style] alphabetization of elements

wjwwood committed:
navfn: fix deprecation warning from pcl-1.7

wjwwood committed:
navfn: update to reflect changes in pcl-1.7

wjwwood committed:
robot_pose_ekf: remove roslib references

wjwwood committed:
robot_pose_ekf: [style] PEP8 changes

wjwwood committed:
voxel_grid: remove unused roslib dependency

wjwwood commented:
@hershwg ok, this should be much easier to review now. And if you don't like or agree with some of the style changes I can cherry pick them out.

wjwwood committed:
remove unused mainpage.dox files (these are rosbuild only)

wjwwood commented:
I just added a commit which removes the rosbuild only `mainpage.dox` files.

wjwwood commented:
I am going to fixup the `costmap_2d` tests as well.

wjwwood committed:
amcl: wrap tests in CATKIN_ENABLE_TESTING check

wjwwood committed:
costmap_2d: fixup tests

This includes using the catkin_add_gtest macro,
using catkin_download_test_data, configuring tests
only if CATKIN_ENABLE_TESTING is set, and ensuring
rostests looked in the correction location for
files which were downloaded.

wjwwood committed:
costmap_2d: changed a function to take a pointer

This fixes a warning from clang:

```
/Users/william/nav_ws/nav_only_ws/src/navigation/costmap_2d/src/costmap_2d_publisher.cpp:48:11: warning: initializing pointer member 'node' with the stack address of parameter
      'ros_node' [-Wdangling-field]
    node(&ros_node), costmap_(costmap), global_frame_(global_frame), active_(false), always_send_full_costmap_(always_send_full_costmap)
          ^~~~~~~~
/Users/william/nav_ws/nav_only_ws/src/navigation/costmap_2d/include/costmap_2d/costmap_2d_publisher.h:95:20: note: pointer member declared here
  ros::NodeHandle* node;
                   ^
1 warning generated.
```

wjwwood committed:
navfn: add missing dependency on netpbm for tests

wjwwood committed:
navfn: netpbm is include namespaced on OS X

wjwwood commented:
I added some more commits which fix testing problems in `amcl`, `costmap_2d`, and `navfn`.

I can run all of the tests for navigation (and they pass):

```
% catkin_make run_tests
Base path: /Users/william/nav_ws/nav_only_ws
Source space: /Users/william/nav_ws/nav_only_ws/src
Build space: /Users/william/nav_ws/nav_only_ws/build
Devel space: /Users/william/nav_ws/nav_only_ws/devel
Install space: /Users/william/nav_ws/nav_only_ws/install
####
#### Running command: "make cmake_check_build_system" in "/Users/william/nav_ws/nav_only_ws/build"
####
Forcing "-j1" for running unit tests.
####
#### Running command: "make run_tests -j1" in "/Users/william/nav_ws/nav_only_ws/build"
####
[  0%] Built target geometry_msgs_generate_messages_cpp
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target costmap_2d_generate_messages_cpp
[  0%] Built target costmap_2d_gencpp
[  2%] Built target voxel_grid
[ 13%] Built target costmap_2d_gencfg
[ 31%] Built target costmap_2d
[ 36%] Built target navfn
[ 42%] Built target path_calc_test
[ 44%] Built target image_loader
[ 47%] Built target utest
[ 50%] Built target voxel_grid_tests
[ 52%] Built target base_local_planner_generate_messages_cpp
[ 52%] Built target base_local_planner_gencpp
[ 52%] Built target base_local_planner_gencfg
[ 86%] Built target base_local_planner
[ 89%] Built target trajectory_planner_ros
[100%] Built target base_local_planner_utest
[100%] Built target line_iterator
[100%] Built target tests
Scanning dependencies of target clean_test_results
[100%] Built target clean_test_results
Scanning dependencies of target _run_tests_navfn_gtest_path_calc_test
-- run_tests.py: execute commands
  /Users/william/nav_ws/nav_only_ws/devel/lib/navfn/path_calc_test --gtest_output=xml:/Users/william/nav_ws/nav_only_ws/build/test_results/navfn/gtest-path_calc_test.xml
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from PathCalc
[ RUN      ] PathCalc.oscillate_in_pinch_point
readPGM() Reading costmap file /Users/william/nav_ws/nav_only_ws/src/navigation/navfn/test/willow_costmap.pgm
readPGM() Size: 1132 x 1217
readPGM() Found 1228386 obstacle cells, 0 free cells, 0 unknown cells
[       OK ] PathCalc.oscillate_in_pinch_point (80 ms)
[ RUN      ] PathCalc.easy_nav_should_always_work
readPGM() Reading costmap file /Users/william/nav_ws/nav_only_ws/src/navigation/navfn/test/willow_costmap.pgm
readPGM() Size: 1132 x 1217
readPGM() Found 1228386 obstacle cells, 0 free cells, 0 unknown cells
[       OK ] PathCalc.easy_nav_should_always_work (41 ms)
[----------] 2 tests from PathCalc (122 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (122 ms total)
[  PASSED  ] 2 tests.
-- run_tests.py: verify result "/Users/william/nav_ws/nav_only_ws/build/test_results/navfn/gtest-path_calc_test.xml"

============ SOME CONTENT REMOVED ============

[----------] Global test environment tear-down
[==========] 25 tests from 4 test cases ran. (1 ms total)
[  PASSED  ] 25 tests.
-- run_tests.py: verify result "/Users/william/nav_ws/nav_only_ws/build/test_results/base_local_planner/gtest-base_local_planner_utest.xml"
[100%] Built target _run_tests_base_local_planner_gtest_base_local_planner_utest
Scanning dependencies of target _run_tests_base_local_planner_gtest
[100%] Built target _run_tests_base_local_planner_gtest
Scanning dependencies of target _run_tests_base_local_planner
[100%] Built target _run_tests_base_local_planner
Scanning dependencies of target run_tests
[100%] Built target run_tests

% catkin_test_results build/test_results
Summary: 37 tests, 0 errors, 0 failures
```

wjwwood commented:
@hershwg These are all of the changes I have for this pull request. I might open another one later with more style changes for the `CMakeLists.txt`.

Let me know if you need to discuss any of these changes. `pcl` and `pcl_conversions` are built on the farm, I am fielding another fix for `pcl_ros` right now which should fix it too.

hershwg commented:
Wow, what a lot of work!  Thanks William.
